### PR TITLE
fix stackoverflow when parse plan failed

### PR DESCRIPTION
**Pull request by mychaow**
## Description

For insertPlan with multi sub plan, it will cause stackoverflow when parse plan failed.

### Content3 ...

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [x] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

**Review by neuyilan — APPROVED** [review]
LGTM

**Review by jt2594838 — APPROVED** [review]
Good, but better with some tests.

File: cross-tests/src/test/java/org/apache/iotdb/cross/tests/tools/importCsv/ExportCsvTestIT.java [hunk 1]
```diff
@@ -46,6 +46,22 @@
...
+  };
+
+  private final String[] outputForUnix = {
+    "------------------------------------------",
+    "Starting IoTDB Client Export Script",
+    "------------------------------------------",
+    "Start to export data from sql statement",
+    "successfully",
+  };
+
```
Comment (neuyilan) [inline cross-tests/src/test/java/org/apache/iotdb/cross/tests/tools/importCsv/ExportCsvTestIT.java:64]: Can we combine the `outputForWindows` and `outputForUnix` together? as I see the string arrays are the same.

And the same as others.

**Review by neuyilan — APPROVED** [review]
LGTM

**Pull request merged**